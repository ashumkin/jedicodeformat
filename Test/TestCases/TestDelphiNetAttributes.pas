unit TestDelphiNetAttributes;

{
 Basic test of attributes in Delphi.NET
}
interface
uses
  Classes;

type

  [Serializable]
  TContainer = class(TObject)
  public
  end;
  [Serializable]
  TContainer2 = class(TObject)
  public
  end;
  TContainer3 = class(TObject)
  public
  end;
  [Serializable]
  TContainer4 = class(TObject)
  public
  end;


  [Serializable]
  TJCFConfiguration = class(TObject)
  public
    CommandlineParams: string;
    Style: integer;

    a,b, c: string;

    [NonSerialized]
    foo, bar: integer;
  end;

  TContainer6 = class(TObject)
  public
    [NonSerialized]
    foo, bar: integer;

    foo2, bar2: integer;

    [NonSerialized]
    foo3, bar3: integer;
    foo4, bar4: integer;
  end;

implementation

end.
