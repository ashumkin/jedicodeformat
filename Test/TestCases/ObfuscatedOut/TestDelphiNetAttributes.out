unit testdelphinetattributes;

interface

uses Classes;

type

  [serializable]
  tcontainer = class(TObject)
  public
  end;

  [serializable]
  tcontainer2 = class(TObject)
  public
  end;

  tcontainer3 = class(TObject)
  public
  end;

  [serializable]
  tcontainer4 = class(TObject)
  public
  end;

  [serializable]
  tjcfconfiguration = class(TObject)
  public
    commandlineparams: string;
    style:    integer;
    a, b, c:  string;
    [nonserialized]
    foo, bar: integer;
  end;

  tcontainer6 = class(TObject)
  public
    [nonserialized]
    foo, bar:   integer;
    foo2, bar2: integer;
    [nonserialized]
    foo3, bar3: integer;
    foo4, bar4: integer;
  end;

implementation

end.
 